src/Microsoft.Health.Fhir.CosmosDb/Features/Search/FhirCosmosSearchService.cs

```diff
@@ -68,7 +82,7 @@ public class FhirCosmosSearchService : SearchService
                 RequestContinuation = searchOptions.ContinuationToken,
             };
 
-            if (searchOptions.CountOnly)
+            if (searchOptions.CountOnly || calculateTotalCount)
             {
                 return new SearchResult(
                     (await _fhirDataStore.ExecuteDocumentQueryAsync<int>(sqlQuerySpec, feedOptions, cancellationToken)).Single(),
```

Comment:
The ContinuationToken is specified above. What does that mean when total is specified? For example, let's say the initial count search returned 1000 items and the search returned 100 items. When the user follows the continuation link to get the second page, what would the total be? Is it still 1000 or would it actually be 900?

Reply:
I created #AB70827 to look into this.